server/app/services/program/ProgramDefinition.java

```diff
@@ -487,9 +487,42 @@ public ImmutableList<BlockDefinition> getNonRepeatedBlockDefinitions() {
         .collect(ImmutableList.toImmutableList());
   }
 
+  /**
+   * Returns a list of the question definitions that may be used to define visibility predicates on
+   * the block definition with the id {@code blockId}.
+   *
+   * <p>The available question definitions for predicates satisfy ALL of the following:
+   *
+   * <ul>
+   *   <li>In a block definition that comes sequentially before the given block definition.
+   *   <li>In a block definition that either has the same enumerator ID as the given block
+   *       definition, or has the same enumerator ID as some "parent" of the given block definition.
+   *   <li>Is not an enumerator.
```

Comment:
suggestion (non-blocking): if it's not too much trouble, could you add and implement "Is not a static text question"?

Reply:
That should be separate from this refactor as it's unrelated to it.